### PR TITLE
[BugFix] Fix parquet footer not have min max statistics caused inaccurate query results

### PR DESCRIPTION
**Pull request by zenoyang**
## Why I'm doing:
parquet footer not have min max statistics caused inaccurate query results.

parquet:
image

debug:
企业微信截图_5d3499de-5896-4ca1-b004-a82b6d6c6ce6

企业微信截图_ba043615-bda7-4ca0-8a2b-ba01830cd3ed

企业微信截图_24fe13e3-68fd-4876-a5d0-aab56e28d890

## What I'm doing:
When statistics is empty, `column_meta->__isset.statistics` is still true, but `statistics.__isset.xxx` may be false, so the value of `statistics.__isset.xxx` needs to be determined.

image

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

**Review by zombee0 — COMMENTED** [review]
LGTM， but which writer did you use, why the statistics is set but there is no statistics, could you add some ut?

**Comment by zenoyang** [comment]
> LGTM， but which writer did you use, why the statistics is set but there is no statistics, could you add some ut?

企业微信截图_4ec2fb26-f4dd-490c-ab6c-22116227dd46

When stats is greater than 4k, statistics is empty, but its isset is always true.
Regarding UT, I will see how to add it.

**Comment by zenoyang** [comment]
> LGTM， but which writer did you use, why the statistics is set but there is no statistics, could you add some ut?

Hi，UT has been added, please help review. @zombee0

File: be/test/formats/parquet/file_reader_test.cpp [hunk 1]
```diff
@@ -3199,4 +3202,46 @@ TEST_F(FileReaderTest, TestTime) {
...
+            {"attr_value", type_varchar},
+            {""},
+    };
+
+    ctx->tuple_desc = Utils::create_tuple_descriptor(_runtime_state, &_pool, slot_descs);
+    Utils::make_column_info_vector(ctx->tuple_desc, &ctx->materialized_columns);
+    ctx->scan_range = (_create_scan_range(_file_no_min_max_stats_path));
+
+    // attr_value = '2'
+    _create_string_conjunct_ctxs(TExprOpcode::EQ, 0, "2", &ctx->conjunct_ctxs_by_slot[0]);
```
Comment (zombee0) [inline be/test/formats/parquet/file_reader_test.cpp:3225]: to cover your case, you need add the predicate to min_max_conjunct

**Review by zombee0 — APPROVED** [review]

**Comment by zombee0** [comment]
the ut issue is the case just set min_max_conjuncts and it didn't work for the miss of statistics, you also need set the conjunct_ctxs_by_slot.

**Review by dirtysalt — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**